### DISCUSSION TITLE
How to get position of the Moon's center?

### DISCUSSION THREAD

**Discussion by Angew** [skyfielders/python-skyfield#802] [Q&A] [answered]
Hi! First off, thanks so much for Skyfield. I'm a total stranger to astronomy and this package is enabling me to turn my dream world-building project into reality.

My situation (to prevent this from being an XY problem): I am trying to use the Moon's position to _roughly_ estimate the time of high & low tides on a particular point on Earth. My naive understanding (based on https://moon.nasa.gov/moon-in-motion/tides/) is that when "Earth's center : my position : Moon center" form a line when projected to Earth's equatorial plane, there's high tide at my position (ignoring the 50' gap for now, and ignoring other instances of high&low tides). I understand this is equivalent to "my position's longitude is the same as one that would be returned by `wgs84.latlon_of(moon_center)`".

**My question**: How do I find the longitude on Earth that is directly beneath the Moon's center at a certain point in time? That is, I am looking for `wgs84.latlon_of(moon_center)` and don't know how to get `moon_center`. I can do this:
```py
eph = Loader.load("de406.bsp")
earth = eph["Earth"]
moon = eph["Moon"]
m = earth.at(my_time).observe(moon)
point = wgs84.latlon_of(m)
```
However, this makes `point` an entire array of lat+lon pairs. I assume that's because the Moon is not a point and thus there's an _area_ beneath it, not a point. I am at a loss at how to get from "the position of the entire Moon" to "the position of the Moon's center." Maybe it's just a trivial step I am missing, but I am in totally unfamiliar waters with these calculations, and I don't really know how to approach it. I would be grateful for any pointers.

(If there's a better approach to get a rough estimate of tides, I would definitely be interested in that as well)

Thanks in advance.

**Comment by psbaltar** [comment, marked as answer]
I'm not sure that tides can be calculated just by knowing the position of the moon. But to answer your specific question, what you have looks to be on the right track. `m` should be the distance between the center of Earth and the center of the Moon. There's a few different latlon functions and I'm not sure what the exact differences are, but I would have used `subpoint_of`. It will return the location that is directly under a given position. e.g. `wgs84.subpoint_of(m)` should be the location on Earth where the Moon will be directly overhead.

https://rhodesmill.org/skyfield/api-topos.html#skyfield.toposlib.Geoid.subpoint_of

If you're getting an array of locations, you might have defined `my_time` incorrectly. Most Skyfield functions can accept a time array and return positions for each time. So if `my_time` is an array, `m` would be an array of the Moon's position at each time.

**Reply by Angew** [reply]
Thanks, you put me on the right track. In my real code, the `latlon_of` call is inside the goal function of a `find_minima` call, and it didn't occur to me that Skyfield could be calling it with an entire array of times (serves me right for not creating an actual SSCCE to ask a question!). Now that I've realised that, I was able to perform the correct computation there.
